01_fib/undrewb/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fib(7)
+}
+
+func sign(n int) int {
+	var result = 1
+	if n < 0 {
+		result = int(math.Pow(-1, float64(n+1)))
```

Comment:
You can optimize this to sth like:
```
[2]int{1,-1}[i%2]
```

Reply:
Nice one. At first glance it feels less readable but the more I look at it the more I like it as it properly captures whats going on.

Reply:
You can make things faster, avoiding the branch and the array lookup at the expense of readability (would definitely warrant an explanatory comment):
```go
func sign(n int) int {
	const shift = 8*unsafe.Sizeof(0) - 1
	return (n+1)<<shift&n>>(shift-1) | 1
}
```
If `sign` is only called for n < 0, you can omit `&n`.

Reply:
Not sure I want to get in to gofu while I am learning!. I'll stick with the readable suggestion from n0npax for now, and put the super fast one on my 'cool stuff' to look at list.